### PR TITLE
Implements `QuizPreview` and `QuestionWrapper` factory

### PR DESCRIPTION
**Pull request by chrichri17**
This PR is doing 2 things:
- create a `QuizPreview` component to display a quiz description and general information (see #32)
- implement a factory pattern to use the right component depending on the `QuestionKind`.

## Usage
To add a new Question: 
- update `schemas/interface.ts`
- then implement the corresponding component in `frontend/scr/components/questions`
- finally register your component in `frontend/scr/components/questions/QuestionWrapper.vue`

## What's next 
- [x] Implements each question component
- [x] The current usage can be tedious since we have an additional step (registering the question component). As a nice to have, we can do this dynamically by loading automatically every files in `frontend/scr/components/questions` except `QuestionWrapper` in the factory.

## Notes
I am not a Vuejs expert, so there might be a better way to do it

### CODE REVIEW

**Comment by chrichri17** [comment]
Current dependencies on/for this PR:
* main
  * **PR #39** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/39" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/YacineSteeve/quizoot/39?utm_source=stack-comment).

File: schemas/interface.ts [hunk 1]
```diff
@@ -217,7 +223,7 @@ declare namespace Quizoot {
         /**
          * All the questions the quiz is made of.
          */
-        questions: QuestionItem[]
+        questions: Question[]
```
Comment (chrichri17) [inline schemas/interface.ts:226]: temporary change until we decide if we're going to use the questions object

File: frontend/src/styles/main.css [hunk 2]
```diff
@@ -25,3 +25,29 @@ body {
     background-color: var(--background-color);
     overflow: hidden;
 }
+
+.btn-group {
```
Comment (YacineSteeve) [inline frontend/src/styles/main.css:29]: I had another suggestion for this, comming in next PR

File: schemas/interface.ts [hunk 3]
```diff
@@ -249,11 +255,11 @@ declare namespace Quizoot {
         /**
          * The quiz author's first name.
          */
-        firstName: string;
+        name: string;
         /**
          * The quiz author's last name. Is optional.
          */
-        lastName?: string;
+        surname?: string;
```
Comment (YacineSteeve) [inline schemas/interface.ts:262]: why 😂

File: schemas/interface.ts [hunk 4]
```diff
@@ -125,6 +125,12 @@ declare namespace Quizoot {
     type OptionId = string;
 
     interface TextQuestion {
+        /**
+         * Placeholder to display in the html `textarea` or `input`
+         * 
+         * @examples ["Type your text here..."]
+         */
+        placeholder?: string;
```
Comment (YacineSteeve) [inline schemas/interface.ts:133]: Good idea

**Review by YacineSteeve — APPROVED** [review]
Nice 💯

File: schemas/interface.ts (hunk 3 above)
Comment (chrichri17) [reply schemas/interface.ts:262]: - First it's not respecting the case style: `lastName` is camelCase while the others fields in the interface are snake_case.
- Second, `name` is shorter and prettier than `firstName`
- Finally, if you really prefer using last and first, then we should write instead `lastname` and `firstname`

File: frontend/src/styles/main.css (hunk 2 above)
Comment (chrichri17) [reply frontend/src/styles/main.css:29]: Awesome ! You are the CSS respo

**Pull request merged**